jme3-terrain/src/main/java/com/jme3/terrain/geomipmap/TerrainGridLodControl.java

```diff
@@ -50,7 +50,7 @@ public TerrainGridLodControl(Terrain terrain, Camera camera) {
     }
     
     @Override
-    protected void updateLOD(List<Vector3f> locations, LodCalculator lodCalculator) {
+    protected void updateLOD(SafeArrayList<Vector3f> locations, LodCalculator lodCalculator) {
```

Comment:
alright checked with Paul, and I'm wrong you need for(Object o:list.getArray())  for it to not create an iterator...